web/federate.go

```diff
@@ -90,7 +90,7 @@ func (h *Handler) federation(w http.ResponseWriter, req *http.Request) {
 		s, wrns, err := q.Select(params, mset...)
 		if wrns != nil {
 			level.Debug(h.logger).Log("msg", "federation select returned warnings", "warnings", wrns)
-			federationErrors.Add(float64(len(wrns)))
+			federationWarnings.Add(float64(len(wrns)))
```

Comment:
Why? Or was the regex too greedy?

Reply:
staticcheck reported correctly that `federationWarnings` was unused. Looking at the [commit](https://github.com/prometheus/prometheus/commit/f0e9196dca9633e77a200be2d8805672f9255cfe#diff-c5eb08a58adae3a6dddb169f8523fc83) that introduced the variable, it looks like there was a typo.

Reply:
Ah, so this actually fixes a bug and we were counting the warnings as errors before? Okay, makes sense!